File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptArrayOptLongValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptArrayOptLongInValue.OptArrayOptLongValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptArrayOptLongValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptArrayOptLongValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                Long[].class,
                OptArrayOptLongValue.class.getDeclaredField("optArrayOptLong").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                Long[].class,
                OptArrayOptLongValue.class.getMethod("getOptArrayOptLong").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptArrayOptLongValue.class,
                OptArrayOptLongValue.class.getMethod("setOptArrayOptLong", Long[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptArrayOptLongValue().getOptArrayOptLong());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptArrayOptLongValue.scala
package com.dslplatform.ocd.values

import OptArrayOptLongInValue.OptArrayOptLongValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptArrayOptLongValue extends SpecExtensions {
  def `Test property field type: Option[Array[Option[Long]]]`() =
    checkType(
      typeOf[Option[Array[Option[Long]]]]
    , typeOf[OptArrayOptLongValue].member("optArrayOptLong": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptArrayOptLongValue().optArrayOptLong === None)
  }
}
